outputting version from incremental-release action for FRAs (#11)
BrightspaceUI/actions: incremental-release/README.md

### Skipping Releases

When a default increment is specified, sometimes you want to bypass it and skip a release. To do this, include `[skip release]` in the commit message.

> ## Publishing a Free-Rang App Using `frau-publisher`

Reviewer: ```suggestion
## Publishing a Free-Range App Using `frau-publisher`
```

A common use case for `incremental-release` is with Free-Range Apps. With each release, FRAs publish to the CDN using [frau-publisher](https://github.com/Brightspace/frau-publisher), which traditionally used Travis CI environment variables to determine the new version.

Here's a sample workflow to publish a Free-Range App:

```yml
name: Release
on:
  push:
    branches:
      - master
jobs:
  release:
    if: "!contains(github.event.head_commit.message, 'skip ci')"
    timeout-minutes: 2
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: Brightspace/third-party-actions@actions/checkout
        with:
          persist-credentials: false
      - name: Setup Node
        uses: Brightspace/third-party-actions@actions/setup-node
      # additional build/validation steps can be run here
      - name: Incremental Release
        id: release
        uses: BrightspaceUI/actions/incremental-release@master
        with:
          GITHUB_TOKEN: ${{ secrets.D2L_GITHUB_TOKEN }}
      - name: Publish
        if: steps.release.outputs.VERSION != ''

>         run: npx frau-publisher --v="${{ steps.semantic-release.outputs.VERSION }}" --f="./dist/**/*.*" --m="app" --t="my-fra"

Reviewer: Do we want to add `env:` and the AWS creds to this example (https://github.com/Brightspace/ifrau/blob/master/.github/workflows/release.yml#L37-L40)?  Was thinking we'd then link to the documentation about setting those up (rather than explaining it here), but that documentation doesn't really exist yet and I'm not sure where it should live.  A page in the guide specifically for migrating FRAs?

Author: That's a good idea -- I'll add them with a [link to here](https://github.com/Brightspace/iam-build-tokens/blob/master/docs/howto-cdn-users.md)

```

Notes:

> * `id` has been to the release step so that we can reference its `VERSION` output parameter in the subsequent step

Reviewer: ```suggestion
* `id` has been added to the release step so that we can reference its `VERSION` output parameter in the subsequent step
```

This is a guess at what was meant 😛

        with:
          persist-credentials: false
      - name: Setup Node
        uses: Brightspace/third-party-actions@actions/setup-node
      # additional build/validation steps can be run here
      - name: Incremental Release
        id: release
        uses: BrightspaceUI/actions/incremental-release@master
        with:
          GITHUB_TOKEN: ${{ secrets.D2L_GITHUB_TOKEN }}
      - name: Publish
        if: steps.release.outputs.VERSION != ''
        run: npx frau-publisher --v="${{ steps.semantic-release.outputs.VERSION }}" --f="./dist/**/*.*" --m="app" --t="my-fra"
        env:
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
```

Notes:
* `id` has been added to the release step so that we can reference its `VERSION` output parameter in the subsequent step
* The publish step will be skipped if no version increment occurred
* Obviously the `--f` (files), `--m` (module type) and `--t` (target directory) parameters passed to `frau-publisher` may be different for your FRA

> * This example uses IAM tokens to publish to the CDN's S3 bucket -- [learn more about setting p IAM tokens](https://github.com/Brightspace/iam-build-tokens/blob/master/docs/howto-cdn-users.md)

Reviewer: ```suggestion
* This example uses IAM tokens to publish to the CDN's S3 bucket -- [learn more about setting up IAM tokens](https://github.com/Brightspace/iam-build-tokens/blob/master/docs/howto-cdn-users.md). Also, when registering your repo in https://github.com/Brightspace/iam-build-tokens/tree/master/terraform/roles we recommend [using a hub role](https://github.com/Brightspace/iam-build-tokens/blob/master/docs/howto-hub-roles.md), so that you can easily add more roles as needed (for example, to setup up visual-diff tests in the future).
```

Author: Good info thx
